staging/src/k8s.io/kubectl/pkg/kuberc/marshal.go

```diff
@@ -65,7 +63,9 @@ func decodePreference(kubercFile string) (*config.Preference, error) {
 			if lenientDecodeErr != nil {
 				// both strict and lenient failed
 				// verbose log the error with the most information about this item and continue
-				klog.V(5).Infof("kuberc: strict decoding error for entry %d in %s: %v", attemptedItems, kubercFile, strictDecodeErr)
+				if verbosity >= 5 {
```

Comment:
This is rather unexpected, I was really hoping we'll be able to initiate the klog appropriately and use it in the early stages of kubectl creation. Have you tried that?

Reply:
Honestly I didn't. But I definitely should. Let me first try to initialize klog first and make changes accordingly.

Reply:
I spent some time on this today and have figured out what is happening. Actually there is no problem about the initialization of the klog. 

The problem is `kubectl/cmd` is not supposed to parse any flags until command execution. So that when we pass `-v=9`, this is not parsed to a klog flag and simply ignored (klog's default verbosity level is 0). This works within command execution, because all the args are mapped to a flag values including the klog's verbosity level at that stage.

Options come to my mind are;
* We continue this ugly manual flag parsing (as we have to do with `kuberc` flag)
* Remove klog logging in that stage, because it doesn't work 

or any other that I couldn't find.

Reply:
cc'ing @liggitt just in case he has some suggestions. I know that we print some valuable logs to troubleshoot any issues about kuberc via `-v=5`.

Reply:
> Options come to my mind are;
>     * We continue this ugly manual flag parsing (as we have to do with `kuberc` flag)
>     * Remove klog logging in that stage, because it doesn't work

That's roughly what I had in mind, we'd manually parse the level and init the logging. Although [your recent comment](https://github.com/kubernetes/kubectl/issues/1731#issuecomment-2842042672) mentioned you weren't successful with that. As mentioned in that issue, I'll try to take a stab at it later this week.

Reply:
Thank you. I tried to initiate similar to this https://github.com/kubernetes/kubernetes/blob/dc992adad385ab631e4a528ee6a342ea71e7a379/staging/src/k8s.io/component-base/cli/run.go#L125 but it was no effect.

Reply:
@ardaguclu have a look at https://github.com/kubernetes/kubernetes/pull/131668 I think I was able to figure this one out :)